kubernetes-crds/pkg/apis/projectriff.io/v1alpha1/windowing.go

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v1alpha1
+
+type Windowing struct {
```

Comment:
these 3 are all mutually exclusive right? how is that handled. Would it be better to have a "Type" enum and a value?

Reply:
@ericbottard For the record, I'm wondering the same thing as @markfisher. You are simulating a union here.  There is presumably quite a lot of overlap between the "branches" of the simulated union for the `Volume` spec, but here the data seems to be entirely disjoint.